mdoc-js/src/main/scala/mdoc/modifiers/ScalajsWorker.scala

```diff
@@ -0,0 +1,106 @@
+package mdoc.modifiers
+
+import org.scalajs.linker.interface._
+import java.net.{URL, URLClassLoader}
+import scala.concurrent.Future
+import java.nio.file.{Path, Paths}
+import scala.concurrent.ExecutionContext
+import scala.language.reflectiveCalls
+
+class ScalajsWorkerState(
+    cachedIRFiles: Seq[IRFile],
+    linker: ClearableLinker
+)
+
+class ScalajsWorker(compilerClasspath: Array[URL], claspath: Array[URL]) {
+  private val shared = getClass().getClassLoader()
+
+  private final class FilteringClassLoader(parent: ClassLoader) extends ClassLoader(parent) {
```

Comment:
I copied it from the LinkerImpl above, but using this doesn't work and brings back to same issue:

```
error: mdoc:js exception
mdoc.internal.markdown.ModifierException: mdoc:js exception
Caused by: org.scalajs.ir.IRVersionNotSupportedException: Failed to deserialize a file compiled with Scala.js 1.8 (supported up to: 1.7): org/scalajs/linker/runtime/RuntimeLong.sjsir
	at org.scalajs.linker.interface.unstable.IRFileImpl$$anonfun$withPathExceptionContext$1.applyOrElse(IRFileImpl.scala:62)
	at org.scalajs.linker.interface.unstable.IRFileImpl$$anonfun$withPathExceptionContext$1.applyOrElse(IRFileImpl.scala:58)
	at scala.util.Failure.recover(Try.scala:233)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:451)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: org.scalajs.ir.IRVersionNotSupportedException: This version (1.8) of Scala.js IR is not supported. Supported versions are up to 1.7
	at org.scalajs.ir.VersionChecks.checkSupported(ScalaJSVersions.scala:77)
	at org.scalajs.ir.Serializers$Deserializer.readHeader(Serializers.scala:1010)
	at org.scalajs.ir.Serializers$Deserializer.deserializeEntryPointsInfo(Serializers.scala:969)
	at org.scalajs.ir.Serializers$.deserializeEntryPointsInfo$$anonfun$1(Serializers.scala:52)
	at org.scalajs.ir.Serializers$.withBigEndian(Serializers.scala:67)
	at org.scalajs.ir.Serializers$.deserializeEntryPointsInfo(Serializers.scala:52)
	at org.scalajs.linker.standard.MemIRFileImpl.$anonfun$entryPointsInfo$1(MemIRFile.scala:31)
	at org.scalajs.linker.standard.MemIRFileImpl.$anonfun$withBuffer$1(MemIRFile.scala:39)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:672)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:431)
```

I am certainly holding this wrong, and would love some input as to how to do it right.